app/src/main/java/org/gdg/frisbee/android/eventseries/GdgEventListFragment.java

```diff
@@ -67,13 +68,18 @@ public void failure(RetrofitError error) {
             App.getInstance().getModelCache().getAsync(cacheKey, false, new ModelCache.CacheListener() {
                 @Override
                 public void onGet(Object item) {
-                    ArrayList<Event> events = (ArrayList<Event>) item;
 
-                    mAdapter.addAll(events);
-                    setIsLoading(false);
-                    Snackbar snackbar = Snackbar.make(getView(), R.string.cached_content,
-                            Snackbar.LENGTH_SHORT);
-                    ColoredSnackBar.info(snackbar).show();
+                    if (checkValidCache(item)) {
+                        ArrayList<Event> events = (ArrayList<Event>) item;
+
+                        mAdapter.addAll(events);
+                        setIsLoading(false);
```

Comment:
You are right. I wish you didn't merge the PR. :smile: I will push the fix to develop branch. Please have a look.